graphsdk/src/androidTest/java/com/microsoft/graph/concurrency/DefaultExecutorsTests.java

```diff
@@ -41,35 +40,26 @@ public void run() {
         });
         // sleep 1 second for background task
         SystemClock.sleep(1000);
+        new SimpleWaiter().signal();
+        assertFalse(Looper.getMainLooper().isCurrentThread());
         assertTrue(success.get());
         assertEquals(1,mLogger.getLogMessages().size());
-        assertTrue(mLogger.isExist(expectedLogMessage));
+        assertTrue(mLogger.hasMessage(expectedLogMessage));
     }
 
     public void testPerformOnForegroundWithResult() {
         String expectedResult = "result value";
         String expectedLogMessage = "Starting foreground task, current active count:0, with result result value";
-        final AtomicBoolean success = new AtomicBoolean(false);
-        final AtomicBoolean failure = new AtomicBoolean(false);
-        ICallback<String> callback = new ICallback<String>() {
-            @Override
-            public void success(String s) {
-                success.set(true);
-                callbackResult = s;
-            }
-
-            @Override
-            public void failure(ClientException ex) {
-                failure.set(true);
-            }
-        };
+        MockCallback callback = new MockCallback();
         defaultExecutors.performOnForeground(expectedResult,callback);
         SystemClock.sleep(1000);
-        assertTrue(success.get());
-        assertFalse(failure.get());
-        assertEquals(expectedResult, callbackResult);
+        new SimpleWaiter().signal();
+        assertFalse(Looper.getMainLooper().isCurrentThread());
+        assertTrue(callback.getSuccess());
+        assertFalse(callback.getFailure());
+        assertEquals(expectedResult, callback.getResult());
         assertEquals(1,mLogger.getLogMessages().size());
-        assertTrue(mLogger.isExist(expectedLogMessage));
+        assertTrue(mLogger.hasMessage(expectedLogMessage));
     }
 
     public void testPerformOnForegroundWithProgress() {
```

Comment:
Use ExecutorTestCallback

---
In reply to: [66116343](https://github.com/microsoftgraph/msgraph-sdk-android/pull/17#discussion_r66116343) [](ancestors = 66116343)